content/en/developers/authorization/oauth2_endpoints.md

```diff
@@ -0,0 +1,234 @@
+| `grant_type`                         | The grant type should be `authorization_code` to receive your initial access token and refresh token, and the grant type should be `refresh_token` to receive any subsequent access and refresh tokens. |
+| `code_verifier` (if PKCE is enabled) | The raw [code verifier][6] used to derive the code challenge sent in the authorization requests.                                                                                                         |
+| `code`                               | The authorization code generated and returned from the previous authorization POST request.                                                                                                         |
+
+#### Example Request 
+
+Use this cURL command to make an access token request:
+
+```
+curl -X POST \
+    -d "grant_type=authorization_code&client_id=$CLIENT_ID
+    client_secret=$CLIENT_SECRET&redirect_uri=$REDIRECT_URI
+    code_verifier=$CODE_VERIFIER&code=$CODE" \
+    "https://api.datadoghq.com/oauth2/v1/token"
+```
+
+#### Success Response
+
+If the access token request is valid and authorized, the [token response][7] returns a `200 OK` status code with the access token and refresh token contained in the body of the HTTP response. 
+
+#### Error Response
+
+Failed requests made to token endpoints should be handled by the application, such as redirecting users to an appropriate error page on the application. 
+
+If a confidential client with an issued client secret makes a token request without providing the `client_secret` parameter, a `401 Unauthorized` status code is returned.
+
+If a token request fails for other reasons, such as a malformed request or an invalid authorization code, a `400 Bad Request` status code (unless specified otherwise) is returned with an [`error`][8] parameter. 
+
+## Revoke tokens 
+
+### `POST /oauth2/v1/revoke`
+
+#### Overview
+
+Users can revoke access or refresh tokens at any time. When revoked, tokens can no longer be used to access Datadog APIs. To revoke a given token, your application makes a POST request to Datadog’s [token revocation endpoint][9]. 
+
+#### Request
+The [revocation request][10] is made with the following parameters in the **body** of the `HTTP POST` request with the `application/x-www-form-urlencoded` format:
+
+| HTTP Body Parameter          | Description                                                                                                              |
+|------------------------------|--------------------------------------------------------------------------------------------------------------------------|
+| `client_id`                    | The client ID of your OAuth2 client.                                                                                      |
+| `client_secret` (if issued)                    | The client secret of your OAuth2 private client.                                                                                       |
```

Comment:
Same as other comment: I'm thinking maybe we should remove the word private here. It might be confusing since we talk about private vs. published client, but this I think is referring to `confidential` client. If that is the case, we can switch it to `confidential`